src/filtering/filterDialogue.component.tsx

```diff
@@ -0,0 +1,128 @@
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  Divider,
+  Grid,
+  Typography,
+} from '@mui/material';
+import React from 'react';
+import FilterInput from './filterInput.component';
+import { useAppDispatch, useAppSelector } from '../state/hooks';
+import {
+  changeAppliedFilters,
+  selectAppliedFilters,
+} from '../state/slices/filterSlice';
+import { Token } from './filterParser';
+import { useChannels } from '../api/channels';
+
+interface FilterDialogueProps {
+  open: boolean;
+  onClose: () => void;
+}
+
+const Heading = (props: React.ComponentProps<typeof Typography>) => (
+  <Typography
+    variant="body1"
+    component="h3"
+    gutterBottom
+    sx={{ fontWeight: 'bold' }}
+  >
+    {props.children}
+  </Typography>
+);
+const Body = (props: React.ComponentProps<typeof Typography>) => (
+  <Typography variant="body2" gutterBottom>
+    {props.children}
+  </Typography>
+);
+
+const FilterDialogue = (props: FilterDialogueProps) => {
+  const { open, onClose } = props;
+  const dispatch = useAppDispatch();
+  const appliedFilters = useAppSelector(selectAppliedFilters);
+  const [filters, setFilters] = React.useState<Token[][]>(appliedFilters);
+  const [errors, setErrors] = React.useState<string[]>(['']);
+  const { data: channels } = useChannels({
+    select: (channels) => {
+      return (
+        channels
+          // TODO: I think it makes sense that we can only apply filters to scalar channels, but should check
+          // should also check we want friendly names here instead of system names
+          .filter((channel) => channel.channel_dtype === 'scalar')
+          .map(
+            (channel) =>
+              ({
+                type: 'channel',
+                value: channel.systemName,
+                label: channel?.userFriendlyName ?? channel.systemName,
+              } as Token)
+          )
+      );
+    },
+  });
+
+  React.useEffect(() => {
+    setFilters(appliedFilters);
+    setErrors(['']);
+  }, [appliedFilters]);
+
+  const handleChangeValue = React.useCallback(
+    (value: Token[]) => setFilters([value]),
+    []
+  );
+  const handleChangeError = React.useCallback(
+    (error: string) => setErrors([error]),
+    []
+  );
+
+  return (
+    <Dialog open={open} onClose={onClose} maxWidth="lg" fullWidth>
+      <DialogTitle>Filters</DialogTitle>
+      <DialogContent>
+        <Grid container columnSpacing={2}>
+          <Grid item xs pr={1}>
+            <Heading>Enter filter</Heading>
+            <Grid container item>
+              <FilterInput
+                channels={channels ?? []}
+                value={filters[0]}
+                setValue={handleChangeValue}
+                error={errors[0]}
```

Comment:
Could you explain why only the first index of filters and errors is being passed here? In the FilterInput component, I console.logged the value prop to see how it was built up as the user added tokens. But what is supposed to go in the other indices of the filters object? For example, what do we intend to put in `filters[1]` and so on?

Reply:
Multiple filter inputs is a separate issue - I just paid the groundwork for that here by making them arrays to begin with but just using the first index

Reply:
Ah I thought so, I checked eCat and see they have the same setup